### PR TITLE
doc: genrest: Separate turbo mode logic from rest of code

### PR DESCRIPTION
**Pull request by ulfalizer**
KCONFIG_TURBO_MODE being twisted up in the normal logic made the code
hard to follow and change.

Use a separate write_dummy_index() function for KCONFIG_TURBO_MODE
instead, and add more documentation for it. Also get rid of options.rst
and just write all the symbol link targets directly to the dummy index
file, which is a bit simpler.

As a small piggybacked improvement for default values, make the heading
'default' instead of 'defaults' when there's only one. The menuconfigs
do this too.

### CODE REVIEW

**Comment by ulfalizer** [comment]
This means the index page changes between turbo and non-turbo modes. I don't know if that's a problem though.

**Review by carlescufi — APPROVED** [review]
LGTM
